app/services/notify_service.rb

```diff
@@ -48,7 +48,7 @@ def message?
     end
 
     def from_staff?
-      @sender.local? && @sender.user.present? && @sender.user_role&.overrides?(@recipient.user_role)
+      @sender.local? && @sender.user.present? && @sender.user_role&.overrides?(@recipient.user_role) && @sender.user_role&.highlighted? && @sender.user_role&.can?(*UserRole::Flags::CATEGORIES[:moderation])
```

Comment:
I think this might need a:
```
@sender == Account.representative || ...
```

i.e., always allow the local Instance representative to override this setting.

Reply:
`Account.representative` is a bit of a misnomer and is *not* the contact account, it's the instance actor.

We unfortunately don't have such a neat way to extract the contact account, the handle is stored in textual form in `Setting.site_contact_username` and parsed and the account looked up on demand. We could do this, but it's a bit of a hot path, so I'm not too sure.

Reply:
Oh, right, hmm, I guess we could always cache the result, but yeah, perhaps too hot a code path here.

Reply:
Right, we could do that, but I'm not really comfortable with it from a technical standpoint. Should we move this PR as is, backport it, and investigate allowing the contact account in the future?

Reply:
I think move forwards with it, arguably it could be backported but isn't strictly necessary with 4.3 round the corner.

i think we should investigate a better mechanism for mods to interact with people on behalf of the instance.